specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp

```diff
@@ -3,75 +3,37 @@ import "@typespec/http";
 import "@typespec/versioning";
 
 import "./azure_chat_extensions.tsp";
+import "./azure_content_filtering.tsp";
+import "./chat_messages.tsp";
 import "./common.tsp";
 import "./functions.tsp";
+import "./tools.tsp";
 
 using TypeSpec.Rest;
 using TypeSpec.Http;
 using TypeSpec.Versioning;
 
 namespace Azure.OpenAI;
 
-@doc("A description of the intended purpose of a message within a chat completions interaction.")
-enum ChatRole {
-  @doc("The role that instructs or sets the behavior of the assistant.")
-  system: "system",
-
-  @doc("The role that provides responses to system-instructed, user-prompted input.")
-  assistant: "assistant",
-
-  @doc("The role that provides input for chat completions.")
-  user: "user",
-
-  @doc("The role that provides function results for chat completions.")
-  @added(ServiceApiVersions.v2023_07_01_Preview)
-  function: "function",
-
-  @doc("The role that represents extension tool activity within a chat completions operation.")
-  @added(ServiceApiVersions.v2023_08_01_Preview)
-  tool: "tool",
-}
-
-@doc("A single, role-attributed message within a chat completion interaction.")
-model ChatMessage {
-  @doc("The role associated with this message payload.")
-  @projectedName("json", "role")
-  role: ChatRole;
-
-  #suppress "@azure-tools/typespec-azure-core/no-nullable" "we explicitly want a nullable string"
-  @doc("The text associated with this message payload.")
-  @projectedName("json", "content")
-  content: string | null;
-
-  @doc("""
-    The name of the author of this message. `name` is required if role is `function`, and it should be the name of the
-    function whose response is in the `content`. May contain a-z, A-Z, 0-9, and underscores, with a maximum length of
-    64 characters.
-    """)
-  @added(ServiceApiVersions.v2023_07_01_Preview)
-  name?: string;
```

Comment:
OpenAI is adding `name` back to their spec: https://github.com/openai/openai-openapi/issues/118

Reply:
Good eye! I believe we have this matched correctly: System, User, and Assistant messages have the optional "name" property while "Tool" and "Function" do not. That appears to be how the fix landed in their OpenAPI3 YAML file, too.